Packs/AWS_SystemManager/Integrations/AWSSystemManager/AWSSystemManager.py

```diff
@@ -0,0 +1,647 @@
+
+    Args:
+    ----
+        ssm_client: AWS SSM client object for making API requests.
+        args (dict): Command arguments containing filters and parameters.
+            - limit (int, optional): Maximum number of documents to retrieve. Defaults to 50.
+            - next_token (str, optional): Token to retrieve the next set of documents.
+
+    Returns:
+    -------
+        list[CommandResults]: A list of CommandResults containing the documents information.
+        and the next token if exists in the response.
+    """
+    def _parse_documents(documents: list[dict[str, Any]]) -> list[dict[str, Any]]:
+        return [
+            {
+                "Name": document.get("Name"),
+                "Display Name": document.get("DisplayName"),
+                "Owner": document.get("Owner"),
+                "Document version": document.get("DocumentVersion"),
+                "Document type": document.get("DocumentType"),
+                "Created date": document.get("CreatedDate"),
+                "Tags": document.get("Tags"),
+                "Platform types": document.get("PlatformTypes"),
+            } for document in documents]
+    kwargs = {"MaxResults": arg_to_number(args.get("limit", 50)) or 50}
+    kwargs.update({"NextToken": next_token}) if (next_token := args.get("next_token")) else None
+
+    response = ssm_client.list_documents(**kwargs)
+    response = convert_datetime_to_iso(response)
+    documents = response.get("DocumentIdentifiers", [])
+
+    command_results = []
+    if next_token := response.get("NextToken"):
+        command_results.append(
+            next_token_command_result(next_token, "DocumentNextToken"),
+        )
+
+    command_results.append(
+        CommandResults(
+            outputs=documents,
+            outputs_key_field="Name",
+            outputs_prefix="AWS.SSM.Document",
+            readable_output=tableToMarkdown(
+                name="AWS SSM Document",
+                t=_parse_documents(documents),
+                json_transform_mapping={
+                    "Tags": JsonTransformer(
+                        is_nested=True,
+                    ),
+                },
+            ),
+        ),
+    )
+    return command_results
+
+
+def test_module(ssm_client) -> str:
+    ssm_client.list_associations(MaxResults=1)
+    return "ok"
+
+
+def main():
+    params = demisto.params()
+    command = demisto.command()
+    args = demisto.args()
+    verify_certificate = not params.get("insecure", True)
+
+    aws_default_region = params["defaultRegion"]
+    aws_role_arn = params.get("roleArn")
+    aws_role_session_name = params.get("roleSessionName")
+    aws_role_session_duration = params.get("sessionDuration")
+    aws_access_key_id = dict_safe_get(params, ["credentials", "identifier"])
+    aws_secret_access_key = dict_safe_get(params, ["credentials", "password"])
+    aws_role_policy = None  # added it for using AWSClient class without changing the code
```

Comment:
Instead of assigning a variable, pass it to AWSClient explicitly (`aws_role_policy=None`)